validator/client/validator.go

```diff
@@ -27,24 +32,61 @@ type validator struct {
 func (v *validator) Initialize(ctx context.Context) {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "validator.Initialize")
 	defer span.Finish()
-
-	cfg := params.BeaconConfig()
-	v.ticker = slotticker.GetSlotTicker(cfg.GenesisTime, cfg.SlotDuration)
 }
 
 // Done cleans up the validator.
 func (v *validator) Done() {
 	v.ticker.Done()
 }
 
+// WaitForChainStart checks whether the beacon node has started its runtime. That is,
+// it calls to the beacon node which then verifies the ETH1.0 deposit contract logs to check
+// for the ChainStart log to have been emitted. If so, it starts a ticker based on the ChainStart
+// unix timestamp which will be used to keep track of time within the validator client.
+func (v *validator) WaitForChainStart(ctx context.Context) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "validator.WaitForChainStart")
+	defer span.Finish()
+	// First, check if the beacon chain has started.
+	stream, err := v.beaconClient.WaitForChainStart(ctx, &ptypes.Empty{})
+	if err != nil {
+		log.Errorf("Could not setup beacon chain ChainStart streaming client: %v", err)
+		return
+	}
+	for {
+		log.Info("Waiting for beacon chain start log from the ETH 1.0 deposit contract...")
+		chainStartRes, err := stream.Recv()
+		// If the stream is closed, we stop the loop.
+		if err == io.EOF {
+			break
+		}
+		// If context is canceled we stop the loop.
+		if ctx.Err() == context.Canceled {
```

Comment:
Can you test these two error conditions? We need the validator to have excellent coverage where all behavior is tested.